aten/src/ATen/native/Loss.cpp

```diff
@@ -273,27 +273,26 @@ Tensor& binary_cross_entropy_out_cpu(const Tensor& input, const Tensor& target,
       .add_owned_const_input(at::squeeze(target))
       .build();
 
-    AT_DISPATCH_FLOATING_TYPES(loss.scalar_type(), "binary_cross_entropy", [&] {
-        at::native::cpu_kernel(
-            iter,
-            [] (scalar_t input_val, scalar_t target_val) {
+    AT_DISPATCH_FLOATING_TYPES_AND(
+        ScalarType::BFloat16, loss.scalar_type(), "binary_cross_entropy", [&] {
```

Comment:
Any reason we don't add fp16 while we are at it? numerical reasons?

Reply:
Sorry, please look at https://github.com/pytorch/pytorch/pull/33206, i am not sure whether the performance of fp16 for CPU is been fix or not.

Reply:
Sorry to bother @Skylion007 again.

I have dived into log1p and log, which make up the binary_cross_entropy operator, and I found that vectorization of them was already supported, so I added "Half" support for binary_cross_entropy on the CPU.